src/atomate2/common/flows/mpmorph.py

```diff
@@ -0,0 +1,528 @@
+        prev_dir: str | Path | None = None,
+        working_outputs: dict[str, Any] | None = None,
+    ) -> Flow:
+        """
+        Run an NVT+EOS equilibration flow.
+
+        Parameters
+        ----------
+        structure : Structure
+            structure to equilibrate
+        prev_dir : str | Path | None (default)
+            path to copy files from
+        working_outputs : dict or None
+            contains the outputs of the flow as it recursively updates
+
+        Returns
+        -------
+        .Flow, an MPMorph flow
+        """
+        if working_outputs is None:
+            if isinstance(self.initial_strain, float | int):
+                self.initial_strain = (
+                    -abs(self.initial_strain),
+                    abs(self.initial_strain),
+                )
+            elif (
+                not isinstance(self.initial_strain, tuple | list | np.array)
+                or len(self.initial_strain) != 2
+            ):
+                raise ValueError(
+                    "`initial_strain` should either be a float, to set "
+                    "a symmetric linear strain of ± `initial_strain`, or a two-element "
+                    "tuple / list to explicitly set linear strain values, "
+                    f"not {self.initial_strain}."
+                )
+
+            linear_strain = np.linspace(
+                *self.initial_strain, self.postprocessor.min_data_points
+            )
+            working_outputs = {
+                "relax": {
+                    key: [] for key in ("energies", "volume", "stress", "pressure")
+                }
+            }
+
+        else:
+            # Fit EOS to running list of energies and volumes
+            self.postprocessor.fit(working_outputs)
+            working_outputs = dict(self.postprocessor.results)
+            for k in ("pressure", "energy"):
+                working_outputs["relax"].pop(k, None)
+
+            # Stop flow here if EOS cannot be fit
+            if (v0 := working_outputs.get("V0")) is None:
+                return Response(output=working_outputs, stop_children=True)
+
+            # Check if equilibrium volume is in range of attempted volumes
+            v0_in_range = (
+                (vmin := working_outputs.get("Vmin"))
+                <= v0
+                <= (vmax := working_outputs.get("Vmax"))
+            )
+
+            # Check if maximum number of refinement NVT runs is set,
+            # and if so, if that limit has been reached
+            max_attempts_reached = len(working_outputs["relax"]["volume"]) >= (
+                (self.max_attempts or np.inf) + self.postprocessor.min_data_points
+            )
+
+            # Successful fit: return structure at estimated equilibrium volume
+            if v0_in_range or max_attempts_reached:
+                final_structure = structure.copy()
+                final_structure.scale_lattice(v0)
+                return final_structure
```

Comment:
What if the return statements returned dictionaries instead of heterogenous outputs? e.g.

A successful return would look like:

```python
            if v0_in_range or max_attempts_reached:
                final_structure = structure.copy()
                final_structure.scale_lattice(v0)
                return {"structure": final_structure, "working outputs": working_outputs}
```

And a failing return:

```python
            if (v0 := working_outputs.get("V0")) is None:
                return Response(output={"working_outputs": working_outputs}, stop_children=True)

```

Advantages:

1. The API more closely mirrors the rest of atomate2 jobs, where we access the structure with `flow.output.structure`

2. By returning the working outputs at the end, there is better insight into how the job is working. I am currently finding the EOS fit is yielding unphysical densities when used with MLIPS. As is, it's a bit hard to collect the child jobs and inspect why the EOS fit is failing. If the working outputs were returned, that'd be very helpful.

3. The outputs can be evolved over time without breaking changes. If someone decided to include additional information in the output, that could be added as a new key:value pair without breaking anything.

Reply:
Good suggestion! Added basically the same, but assigned the output prior to the `stress` and `energy` keys being popped from working outputs (that's the full EOS fit output, should be more useful for debugging)